crates/forge_loader/src/manifest.rs

```diff
@@ -59,25 +89,66 @@ enum Interval {
     Week,
 }
 
+// Thank you to whomeever kept this one the same. T.T
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
 struct ScheduledTrigger<'a> {
     #[serde(flatten, borrow)]
     raw: RawTrigger<'a>,
     interval: Interval,
 }
 
+// compass DataProvider module
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
-struct WorkflowPostFunction<'a> {
+pub struct DataProvider<'a> {
     #[serde(flatten, borrow)]
-    raw: RawTrigger<'a>,
+    key: &'a str,
+    callback: Option<&'a str>,
+}
+
+// Struct for Custom field Module. Check that search suggestion gets read in correctly.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CustomField<'a> {
+    // all attributes below involve function calls
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    value: Option<&'a str>,
+    search_suggestions: Option<&'a str>,
+    edit: Option<&'a str>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct UiModificatons<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct WorkflowValidator<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct WorkflowPostFunction<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
 }
 
+// Add more structs here for deserializing forge modules
 #[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
 pub struct ForgeModules<'a> {
+    // deserializing non user-invocable modules
     #[serde(rename = "macro", default, borrow)]
     macros: Vec<MacroMod<'a>>,
     #[serde(rename = "function", default, borrow)]
     pub functions: Vec<FunctionMod<'a>>,
+    #[serde(rename = "contentByLineItem", default, borrow)]
```

Comment:
```suggestion
    #[serde(rename = "confluence:contentByLineItem", default, borrow)]
```

Also, you're missing these confluence modules:

- [`confluence:contentAction`](https://developer.atlassian.com/platform/forge/manifest-reference/modules/confluence-content-action/#properties)
- [`confluence:contextMenu`](https://developer.atlassian.com/platform/forge/manifest-reference/modules/confluence-context-menu/#properties)
- [`confluence:globalPage`](https://developer.atlassian.com/platform/forge/manifest-reference/modules/confluence-global-page/#properties)
- [`confluence:homepageFeed`](https://developer.atlassian.com/platform/forge/manifest-reference/modules/confluence-homepage-feed/#properties)
- [`confluence:spacePage`](https://developer.atlassian.com/platform/forge/manifest-reference/modules/confluence-space-page/#properties)
- [`confluence:spaceSettings`](https://developer.atlassian.com/platform/forge/manifest-reference/modules/confluence-space-settings/)